### DISCUSSION TITLE
Spring Boot Kafka Consumer Not Processing Messages After Fixing Processing Issue

### DISCUSSION THREAD

**Discussion by hemantmaersk** [spring-projects/spring-kafka#3449] [Q&A]
Hi Spring Boot Kafka Team,

I’m facing an issue with Kafka message processing in our Spring Boot application and would appreciate your assistance. Below are the details of the problem:

**Spring Boot and Kafka Configuration:**
- **Spring Boot Version**: 3.3.0
- **Consumer Configuration**:
  - **Offset Management**: spring.kafka.consumer.auto-offset-reset=earliest
  - **Retry Settings**: [default
  - **Auto-Commit**: Disabled (manual offset commits are used) [spring.kafka.consumer.enable-auto-commit=false]

**Issue Description:**
Our Kafka consumer encountered a failure while processing messages, and after 10 retry attempts, it failed to process the message. We do not have a dead letter topic configured. After fixing the underlying issue and restarting the application, the consumer does not resume processing as expected. Message that failed to process is still pending in input topic.

**Questions:**
1. **Handling Failures and Restarts**: Given that we are using manual offset commits, what is the recommended approach to handle message processing failures and ensure the consumer resumes from the correct offset after a restart? 

<img width="935" alt="image" src="https://github.com/user-attachments/assets/22c4c118-0c5c-483a-866f-c25a6501ee12">
   
2. **Offset Reset**: How can we manually reset the consumer's offset using Spring Kafka to reprocess messages that failed previously? Are there specific methods or configurations in Spring Kafka to achieve this?

3. **Configuration Checks**: Are there any particular consumer configurations or settings in Spring Kafka that we should verify to ensure that the consumer resumes processing correctly after restarting?

4. **Best Practices**: Could you suggest best practices for handling message failures and retries in Spring Kafka, especially when a dead letter topic is not used?

**Request for Verification:**
Please check if this might be an issue with the Spring Kafka or Spring boot or if it’s related to our configuration. If it’s identified as a potential issue with the Spring Kafka or Spring boot library itself, I would be happy to raise a GitHub issue for further investigation.
Thank you for your help!

**Comment by sobychacko** [comment]
Spring Kafka sets `enable.auto.commit` to `false` by default and the default ack mode is `BATCH`. Are you setting a manual ack mode type in your configuration?

Spring Kafka provides a seekable API to seek the processing to a specific offset. Have you looked into that? https://docs.spring.io/spring-kafka/reference/kafka/seek.html#page-title

Can you create a reproducible sample (with a predictable way to reproduce consistently but a very small application) and share it with us? We will be happy to triage the issue further. Without knowing more details about your configuration, code, etc., it is harder to say what the issue is.
